src/pages/flows/[flow].tsx

```diff
@@ -1,20 +1,32 @@
 import Chat from "@pb/components/chat";
 import FlowBuilder from "@pb/components/flow-builder";
+import type { PBNodeProps } from "@pb/components/flow-builder/flow-builder.types";
 import HeadMeta from "@pb/components/head-meta";
 import Header from "@pb/components/header";
-import TwoColumnLayout from "@pb/components/layout/two-column";
+import NodeOptionsForm from "@pb/components/node-options-form";
+import useFlowStore, { flowSelector } from "@pb/store/flow-builder.store";
 import { type NextPage } from "next";
+import { shallow } from "zustand/shallow";
 
 const Flow: NextPage = () => {
+  const { flowMode, selectedNode } = useFlowStore(flowSelector, shallow);
+  const showChat = flowMode === 1;
+  const showConfigs = !!selectedNode && !showChat;
+
   return (
     <>
       <HeadMeta title="PepperBot | Demo" />
       <main className="min-h-screen bg-base-100">
         <Header />
-        <TwoColumnLayout
-          leftContent={<FlowBuilder />}
-          rightContent={<Chat />}
-        />
+        <div className="row flex h-[calc(100vh-65px)] w-screen bg-base-200">
+          <div className="h-full flex-1">
+            <FlowBuilder />
+          </div>
+          {showConfigs ? (
+            <NodeOptionsForm {...(selectedNode as unknown as PBNodeProps)} />
+          ) : null}
+          {showChat ? <Chat /> : null}
+        </div>
       </main>
     </>
   );
```

Comment:
We can simplify the code and minimize it as follows:

```
const Flow: NextPage = () => {
  const { flowMode, selectedNode } = useFlowStore(flowSelector, shallow);
  const showChat = flowMode === 1;

  return (
    <>
      <HeadMeta title="PepperBot | Demo" />
      <Header />
      <main className="min-h-screen bg-base-100">
        <div className="row flex h-[calc(100vh-65px)] w-screen bg-base-200">
          <div className="h-full flex-1">
            <FlowBuilder />
          </div>
          {showChat ? (
            <Chat />
          ) : !!selectedNode ? (
            <NodeOptionsForm {...(selectedNode as unknown as PBNodeProps)} />
          ) : null}
        </div>
      </main>
    </>
  );
};
```

Please note that the `<Header />` component would be better placed outside the main tag. What are your thoughts on moving the header to the _app.tsx component instead of adding it to every new page we create?

Reply:
nested ternaries are hard to read, trust me ;)